### ISSUE TITLE
function-call-in-default-argument (B008) exception, `frozenset`, not treated as exception

### ISSUE THREAD

**Issue by hunterhogan** [astral-sh/ruff#25248]
### Summary

Rule [function-call-in-default-argument (B008)](https://docs.astral.sh/ruff/rules/function-call-in-default-argument/) reads, "Parameters with immutable type annotations will be ignored by this rule."

Python documentation reads, "[The `frozenset` type is immutable.](https://docs.python.org/3/library/stdtypes.html#set-types-set-frozenset)"

However, this triggers B008:

```python
def functionB008(anArgument: frozenset[str] = aFunction()) -> None:
    pass
```

[Playground](https://play.ruff.rs/efaef74f-3450-4578-b334-909ee01db9bb)

### Version

v0.15.13
